sphinx_needs/functions/common.py

```diff
@@ -171,26 +172,32 @@ def copy(
             NeedsSphinxConfig(app.config),
             filter,
             need,
-            location=(need["docname"], need["lineno"]) if need["docname"] else None,
+            location=(need["docname"], need["lineno"])
+            if need and need["docname"]
+            else None,
         )
         if result:
             need = result[0]
 
-    value = need[option]  # type: ignore[literal-required]
+    if need is None:
+        raise ValueError("Need not found")
```

Comment:
if there is a `need_id`, then the `need` would not be `None` anyway 😅
its of note, that these exceptions are caught in the calling func, and the warning given with source information, so this will be more infomative than just this message

Reply:
ok thanks that clears things up.